src/query/plan/operator.cpp

```diff
@@ -5176,26 +5178,28 @@ class CallProcedureCursor : public Cursor {
     // empty result set vs procedures which return `void`. We currently don't
     // have procedures registering what they return.
     // This `while` loop will skip over empty results.
+    auto module = std::shared_ptr<procedure::Module>{};
+    auto proc = static_cast<const mgp_proc *>(nullptr);
+
     while (result_row_it_ == result_->rows.end()) {
-      // It might be a good idea to resolve the procedure name once, at the
-      // start. Unfortunately, this could deadlock if we tried to invoke a
-      // procedure from a module (read lock) and reload a module (write lock)
-      // inside the same execution thread. Also, our RWLock is set up so that
-      // it's not possible for a single thread to request multiple read locks.
-      // Builtin module registration in query/procedure/module.cpp depends on
-      // this locking scheme.
```

Comment:
This no longer applies. The ModulePtr in the past use to lock the registry (this IMO was too strong), hence this comment to avoid deadlock. 

We now prevent replacing query module while it is in use. The shared_ptr's use_count() != 1 means at least one query is using it.

Reply:
Aha yes, removing is prevented due to check